src/UmaCtfAdapter.sol

```diff
@@ -184,13 +186,30 @@ contract UmaCtfAdapter is IUmaCtfAdapter, Auth, BulletinBoard, IOptimisticReques
 
         if (!_isInitialized(questionData)) revert NotInitialized();
         if (_isFlagged(questionData)) revert Flagged();
+        if (questionData.resolved) revert Resolved();
 
-        questionData.emergencyResolutionTimestamp = block.timestamp + emergencySafetyPeriod;
+        questionData.emergencyResolutionTimestamp = block.timestamp + EMERGENCY_SAFETY_PERIOD;
         questionData.paused = true;
 
         emit QuestionFlagged(questionID);
     }
 
+    /// @notice Unflags a market for emergency resolution
+    /// @param questionID - The unique questionID of the question
+    function unflag(bytes32 questionID) external onlyAdmin {
+        QuestionData storage questionData = questions[questionID];
+
+        if (!_isInitialized(questionData)) revert NotInitialized();
+        if (!_isFlagged(questionData)) revert NotFlagged();
+        if (questionData.resolved) revert Resolved();
+        if (block.timestamp > questionData.emergencyResolutionTimestamp) revert SafetyPeriodPassed();
```

Comment:
why do we care to add this constraint?

Reply:
If the emergency resolve timestamp has passed, an admin should not be able to "go back on his word" and unflag the question.

Reply:
not sure if I agree, I feel like that is an arbitrary constraint to check. The safety period protects users, but they might decide well after the period (before they resolve because there is no deadline on resolving after the period passes) to unflag.